src/app/inventory/dimItemInfoService.factory.js

```diff
@@ -59,58 +59,69 @@ export function ItemInfoService(SyncService, $i18next, toaster, $q) {
   }
 
   // Returns a function that, when given an account, returns the item info source for that platform
-  return function(account, destinyVersion = 1) {
-    const key = `dimItemInfo-m${account.membershipId}-p${account.platformType}-d${destinyVersion}`;
+  let key;
+  return {
+    bulkSave: function(items) {
+      getInfos(key).then((infos) => {
+        items.forEach((item) => {
+          infos[`${item.hash}-${item.id}`] = { tag: item.dimInfo.tag };
+        });
+        setInfos(key, infos);
+      });
+    },
+    init: function(account, destinyVersion = 1) {
```

Comment:
@kyleshay I think this could break item infos across destiny versions and platforms. Before, we'd create a new object for each account/platform - now, we create one, and change global state when `init` is called. Why couldn't the `bulkSave` method be put on the object returned from this function instead?

Reply:
Hmm, ok. Is this more along the right path? 

https://github.com/DestinyItemManager/DIM/pull/2390